internal/resources/cluster/resource_cluster.go

```diff
@@ -213,24 +232,87 @@ func flattenSpec(spec *clustermodel.VmwareTanzuManageV1alpha1ClusterSpec) (data
+	value, ok := diff.GetOk(attachClusterKey)
+	if !ok {
+		return nil
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("attach cluster data: %v is not valid: minimum one valid kube config type is required among: %v", data, strings.Join(KubeConfigWayAllowed[:], `, `))
+	}
+
+	kubeConfigData := data[0].(map[string]interface{})
+
+	kubeConfigTypeFound := make([]string, 0)
+
+	if v, ok := kubeConfigData[attachClusterKubeConfigPathKey]; ok {
+		if v1, ok := v.(string); ok && len(v1) != 0 {
+			kubeConfigTypeFound = append(kubeConfigTypeFound, attachClusterKubeConfigPathKey)
+		}
+	}
+
+	if v, ok := kubeConfigData[attachClusterKubeConfigRawKey]; ok {
+		if v1, ok := v.(string); ok && len(v1) != 0 {
+			kubeConfigTypeFound = append(kubeConfigTypeFound, attachClusterKubeConfigRawKey)
+		}
+	}
+
+	if len(kubeConfigTypeFound) == 0 {
+		return fmt.Errorf("no valid kube config type found: minimum one valid kube config type is required among: %v", strings.Join(KubeConfigWayAllowed[:], `, `))
+	} else if len(kubeConfigTypeFound) > 1 {
+		return fmt.Errorf("found kube config types: %v are not valid: maximum one valid kube config type is allowed", strings.Join(kubeConfigTypeFound, `, `))
+	}
+
+	return nil
+}
+
 func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
 	config := m.(authctx.TanzuContext)
 
 	var (
-		kubeconfigfile string
-		k8sclient      k8sClient.Client
-		err            error
+		k8sclient  *k8sClient.Client
+		err        error
+		kubeConfig interface{}
 	)
 
-	if _, ok := d.GetOk(attachClusterKey); ok {
-		kubeconfigfile, _ = d.Get(helper.GetFirstElementOf(attachClusterKey, attachClusterKubeConfigKey)).(string)
+	if v, ok := d.GetOk(attachClusterKey); ok {
+		if v == nil {
+			return diag.Errorf("data for attach cluster block not found: %v", v)
+		}
 
-		if kubeconfigfile != "" {
-			k8sclient, err = getK8sClient(kubeconfigfile)
-			if err != nil {
-				log.Println("[ERROR] error while creating kubernetes client: ", err.Error())
-				return diag.FromErr(err)
+		isKubeConfigPresent := func(typeKey string) bool {
+			if value, ok := d.GetOk(helper.GetFirstElementOf(attachClusterKey, typeKey)); ok {
+				if value != nil {
+					kubeConfig = value
+					return true
+				}
 			}
+
+			return false
+		}
+
+		switch {
+		case isKubeConfigPresent(attachClusterKubeConfigPathKey):
+			kubeConfigFile, _ := kubeConfig.(string)
+			k8sclient, err = getK8sClient(withPath(kubeConfigFile))
+		case isKubeConfigPresent(attachClusterKubeConfigRawKey):
+			rawKubeConfig, _ := kubeConfig.(string)
+			k8sclient, err = getK8sClient(withRaw(rawKubeConfig))
+		}
+
+		if err != nil {
+			log.Println("[ERROR] error while creating kubernetes client: ", err.Error())
+			return diag.FromErr(err)
+		}
+
+		if k8sclient == nil {
+			err = errors.New("k8s client not found")
```

Comment:
check the error message, k8s client is something we are creating, so not found will be a wrong error

Reply:
changed